src/main/java/uk/gov/companieshouse/overseasentitiesapi/validation/AddressDtoValidator.java

```diff
@@ -66,14 +66,15 @@ private boolean validateCounty(String parentAddressField, String county, Errors
                 && StringValidators.isValidCharacters(county, qualifiedFieldName, errors, loggingContext);
     }
 
-    private boolean validateCountry(String parentAddressField, String country, Errors errors, String loggingContext) {
+    private boolean validateCountry(String parentAddressField, String country, List<String> allowedCountries, Errors errors, String loggingContext) {
         String qualifiedFieldName = getQualifiedFieldName(parentAddressField, AddressDto.COUNTRY_FIELD);
-        String oeDueDiligenceAdddressPath = getQualifiedFieldName(OverseasEntitySubmissionDto.OVERSEAS_ENTITY_DUE_DILIGENCE, OverseasEntityDueDiligenceDto.IDENTITY_ADDRESS_FIELD);
-        if (parentAddressField.equalsIgnoreCase(oeDueDiligenceAdddressPath)) {
-            return UkCountry.isValid(country, qualifiedFieldName, errors, loggingContext);
-        } else {
-            return Country.isValid(country, qualifiedFieldName, errors, loggingContext);
+        boolean isOnList = allowedCountries.contains(country);
+        if (!isOnList) {
+            String validationMessage = String.format(ValidationMessages.COUNTRY_NOT_ON_LIST_ERROR_MESSAGE, country);
```

Comment:
Sonar has found a code smell for this line. We should declare this local variable with `var` instead of `String`. 

Note: it seems we are returning boolean value from the following methods but we are not using it, probably in the future we should remove it.

Reply:
👍 will fix. I'll also investigate why these methods aren't void (and change) if return value not used.

Reply:
Fixed this.